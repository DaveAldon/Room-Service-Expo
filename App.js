import React from "react";
import { SafeAreaView, StyleSheet } from "react-native";
import { RoomServiceProvider } from "@roomservice/react";
import { RoomServiceHome } from "./components/RoomServiceHome/index";

async function AuthCheck(params) {
  const response = await fetch("http://172.16.29.51:3000/api/roomservice", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    credentials: "include",
    body: JSON.stringify({
      room: params.room,
      user: CreateGuid(),
    }),
  });

  if (response.status === 401) {
    throw new Error("Unauthorized!");
  }

  const body = await response.json();
  return body;
}

const App = () => {
  return (
    <RoomServiceProvider
      clientParameters={{
        auth: AuthCheck,
      }}
    >
      <SafeAreaView>
        <RoomServiceHome />
      </SafeAreaView>
    </RoomServiceProvider>
  );
};

function CreateGuid() {
  function _p8(s) {
    var p = (Math.random().toString(16) + "000000000").substr(2, 8);
    return s ? "-" + p.substr(0, 4) + "-" + p.substr(4, 4) : p;
  }
  return _p8() + _p8(true) + _p8(true) + _p8();
}

const styles = StyleSheet.create({});

export default App;
