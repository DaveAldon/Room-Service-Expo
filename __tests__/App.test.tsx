import React from "react";
import renderer from "react-test-renderer";
import App from "../App";

describe("<App />", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has safearea child", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree?.type).toEqual("RCTSafeAreaView");
  });

  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
