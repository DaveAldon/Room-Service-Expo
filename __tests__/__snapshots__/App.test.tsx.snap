// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "95%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": "10%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#abffc1",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": "100%",
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text>
          Title Data:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#abffc1",
            "borderRadius": 10,
            "flexDirection": "row",
            "height": "100%",
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text>
          Description Data:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": "90%",
        }
      }
    >
      <Text>
        Loading...
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
